Missing `winsize` when compiling on OmniOS

When I try to build the `main` branch on [OmniOS](https://omnios.org/) , I get the error below.

```
william@omnios:~/Projects/gui/FTXUI/build$ make
[  1%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/box.cpp.o
[  1%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/color.cpp.o
[  2%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/color_info.cpp.o
[  2%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/surface.cpp.o
[  2%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/screen.cpp.o
[  3%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/string.cpp.o
[  3%] Building CXX object CMakeFiles/screen.dir/src/ftxui/screen/terminal.cpp.o
/home/william/Projects/gui/FTXUI/src/ftxui/screen/terminal.cpp: In function 'ftxui::Dimensions ftxui::Terminal::Size()':
/home/william/Projects/gui/FTXUI/src/ftxui/screen/terminal.cpp:286:3: error: 'winsize' was not declared in this scope
  286 |   winsize w{};
      |   ^~~~~~~
/home/william/Projects/gui/FTXUI/src/ftxui/screen/terminal.cpp:287:43: error: 'TIOCGWINSZ' was not declared in this scope
  287 |   const int status = ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);  // NOLINT
      |                                           ^~~~~~~~~~
/home/william/Projects/gui/FTXUI/src/ftxui/screen/terminal.cpp:287:56: error: 'w' was not declared in this scope
  287 |   const int status = ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);  // NOLINT
      |                                                        ^
make[2]: *** [CMakeFiles/screen.dir/build.make:163: CMakeFiles/screen.dir/src/ftxui/screen/terminal.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:441: CMakeFiles/screen.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
william@omnios:~/Projects/gui/FTXUI/build$ 
```

--- a/src/ftxui/screen/terminal.cpp
+++ b/src/ftxui/screen/terminal.cpp
@@ -22,6 +22,9 @@
 #include <sys/ioctl.h>  // for winsize, ioctl, TIOCGWINSZ
 #include <unistd.h>     // for STDOUT_FILENO
 #endif
+#if defined(__sun) || defined(__illumos__)
+#include <sys/termios.h>  // for winsize on illumos
+#endif
 
 namespace ftxui {
 
@@ -108,7 +111,8 @@ struct Quirks::Impl {
 
 Quirks::Quirks() : impl_(std::make_unique<Impl>()) {}
 Quirks::~Quirks() = default;
-Quirks::Quirks(const Quirks& other) : impl_(std::make_unique<Impl>(*other.impl_)) {}
+Quirks::Quirks(const Quirks& other)
+    : impl_(std::make_unique<Impl>(*other.impl_)) {}
 Quirks& Quirks::operator=(const Quirks& other) {
   if (this != &other) {
     *impl_ = *other.impl_;

